shared-worker/firebase-worker.js

```diff
@@ -0,0 +1,54 @@
+importScripts('../firebase/firebase-app.js');
```

Comment:
Maybe add a comment at the top of the file explaining what this file is / how to use it? Something along the lines of

```
/**
 * Dedicated web worker that loads Firebase off of the main browser thread.  
 * Must be loaded from your main page and initialized with the cross-tab coordination shared-worker, e.g.:
 *
 * dedicatedWorker = new Worker("firebase-worker.js");
 * dedicatedWorker.postMessage(
 *   {sharedWorkerPort: sharedWorker.port}, [sharedWorker.port]
 * );
 */
```